soccer/planning/Path.cpp

```diff
@@ -32,4 +37,42 @@ void Path::draw(SystemState* const state, const QColor& color,
     addPoint(end().motion);
 }
 
-}  // namespace Planning
+std::unique_ptr<ConstPathIterator> Path::iterator(RJ::Time startTime,
+                                                  float deltaT) const {
+    return std::move(
+        std::make_unique<ConstPathIterator>(this, startTime, deltaT));
+}
+
+bool Path::pathsIntersect(const std::vector<const Path*>& paths, float* hitTime,
+                          Geometry2d::Point* hitLocation,
+                          RJ::Time startTime) const {
+    const float deltaT = 0.05;
+    const float hitRadius = Robot_Radius * 2.5f;
```

Comment:
I don't know, it seems a bit clunky. What do we need individual robot avoid radius's for?